src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -367,8 +357,8 @@ object DepositHandler {
       }
   }
 
-  private def compareChecksumAgainstReferredBag(file: String, checksum: String, url: String)(implicit id: String, bs: BagStoreBase): Try[Unit] = {
-    val referredFile = getReferredFile(url, bs.baseUrl)
+  private def compareChecksumAgainstReferredBag(file: String, checksum: String, url: String)(implicit id: String, bs: Option[BagStoreSettings]): Try[Unit] = {
+    val referredFile = getReferredFile(url, bs.get.baseUrl)
```

Comment:
Not convinced by your statement. You can still get there with a `None` if you look carefully!
I refactored the code such that you give the `BagStoreSettings` rather than its `Option` here. Makes the code cleaner here! Given that you do not expect this to be happening, I created a `Failure` in https://github.com/vesaakerman/easy-sword2/pull/4/files#diff-ad17b49d50f1b93808fb326e8f79d0d8R252.